src/D2L.CodeStyle.Analyzers/Immutability/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -37,6 +37,38 @@ CompilationStartAnalysisContext context
 				),
 				SymbolKind.NamedType
 			);
+
+			context.RegisterSymbolAction(
+				ctx => AnalyzeMember( ctx, immutabilityContext ),
+				SymbolKind.Field,
+				SymbolKind.Property
+			);
+		}
+
+		private static void AnalyzeMember(
+			SymbolAnalysisContext ctx,
+			ImmutabilityContext immutabilityContext
+		) {
+			// We only care about checking static fields/properties. These
+			// are global variables, so we always want them to be immutable.
+			// The fields/properties of [Immutable] types get handled via
+			// AnalyzeTypeDeclaration.
+			if ( !ctx.Symbol.IsStatic ) {
+				return;
+			}
+
+			// Ignore const things, which include enum names.
+			if ( ctx.Symbol is IFieldSymbol f && f.IsConst ) {

```

Comment:
Can you send xplain this one? I don't think I understand.

Reply:
```cs
enum Foo {
  Bar
}
```

Bar is a static field that is not implicitly declared. Critically it has zero variable declarators, so if passed it to CheckMember it would throw on an invalid cast in some code that looks up syntax locations.

Reply:
Also probably just regular consts but their types are ok